kroxylicious-systemtests/src/test/java/io/kroxylicious/systemtests/TopicEncryptionST.java

```diff
@@ -0,0 +1,127 @@
+    void setUp() {
+        vaultOperator = new Vault(Constants.VAULT_DEFAULT_NAMESPACE);
+        NamespaceUtils.createNamespaceWithWait(Constants.VAULT_DEFAULT_NAMESPACE);
+        vaultOperator.deploy();
+
+        List<Pod> kafkaPods = kubeClient().listPodsByPrefixInName(Constants.KROXY_DEFAULT_NAMESPACE, clusterName);
+        if (!kafkaPods.isEmpty()) {
+            LOGGER.warn("Skipping kafka deployment. It is already deployed!");
+            return;
+        }
+        LOGGER.info("Deploying Kafka in {} namespace", Constants.KROXY_DEFAULT_NAMESPACE);
+
+        Kafka kafka = KafkaTemplates.kafkaPersistentWithKRaftAnnotations(Constants.KROXY_DEFAULT_NAMESPACE, clusterName, 3).build();
+
+        resourceManager.createResourceWithWait(
+                KafkaNodePoolTemplates.kafkaBasedNodePoolWithDualRole(BROKER_NODE_NAME, kafka, 3).build(),
+                kafka);
+    }
+
+    @AfterAll
+    void cleanUp() throws IOException {
+        if (vaultOperator != null) {
+            vaultOperator.delete();
+        }
+    }
+
+    @Test
+    void produceAndConsumeEncryptedMessage(String namespace) {
+        String topicName = "my-topic";
+        String message = "Hello-world";
+        int numberOfMessages = 1;
+        String consumedMessage = message + " - " + (numberOfMessages - 1);
+
+        var testKekManager = new VaultTestKekManager(Constants.VAULT_DEFAULT_NAMESPACE, Constants.VAULT_SERVICE_NAME + "-0");
+        testKekManager.generateKek(topicName);
+
+        // start Kroxylicious
+        LOGGER.info("Given Kroxylicious in {} namespace with {} replicas", namespace, 1);
+        kroxylicious = new Kroxylicious(namespace);
+        kroxylicious.deployPortPerBrokerPlainWithTopicEncryptionFilter(clusterName, 1, topicName);
+        String bootstrap = kroxylicious.getBootstrap();
+
+        LOGGER.info("And KafkaTopic in {} namespace", namespace);
+        KafkaSteps.createTopic(namespace, topicName, bootstrap, 1, 2);
+
+        LOGGER.info("When {} messages '{}' are sent to the topic '{}'", numberOfMessages, message, topicName);
+        KroxyliciousSteps.produceMessages(namespace, topicName, bootstrap, message, numberOfMessages);
+
+        LOGGER.info("Then the {} messages are consumed", numberOfMessages);
+        String kafkaBootstrap = clusterName + "-kafka-bootstrap." + Constants.KROXY_DEFAULT_NAMESPACE + ".svc.cluster.local:9092";
+        String resultEncrypted = KroxyliciousSteps.consumeEncryptedMessages(namespace, topicName, kafkaBootstrap, numberOfMessages, Duration.ofMinutes(2));
+        LOGGER.info("Received: " + resultEncrypted);
+        assertThat("'" + consumedMessage + "' message not consumed!", resultEncrypted.contains(topicName + "vault"));
+    }
+
+    @Test
+    void produceAndConsumeMessage(String namespace) {
+        String topicName = "my-topic2";
+        String message = "Hello-world";
+        int numberOfMessages = 1;
+        String consumedMessage = message + " - " + (numberOfMessages - 1);
```

Comment:
Can we please rename this to `expectedMessage`

Reply:
done